torch/quantization/fx/quantization_patterns.py

```diff
@@ -520,29 +530,31 @@ def convert(self, quantizer: QuantizerCls, node: Node, load_arg: Callable,
                     # output for dynamically quantized linear op is not quantized
                     return op_out
             else:  # non-reference option
-                # linear args
-                # (x, weight, bias, ...)
-                weight_quantized = weight_is_statically_quantized(qconfig)
-                linear_weight = load_arg(quantized=weight_quantized)(self.linear_node.args[1])
-
-                # get other arguments
-                kwargs = {**load_arg(quantized=False)(self.linear_node.kwargs)}
-                # pack weight
-                bias = None
-                # all args after bias, including bias
-                other_args = load_arg(quantized=False)(self.linear_node.args[2:])
-                if len(self.linear_node.args) > 2:
-                    bias = load_arg(quantized=False)(self.linear_node.args[2])
-                    other_args = other_args[1:]  # remove the bias argument
-                else:
-                    assert 'bias' in kwargs, \
-                        'expect bias provided as a keyword argument when it is not a positional argument'
-                    bias = kwargs['bias']
-                    kwargs.pop('bias')
-                prepack_args = (linear_weight, bias)
-                prepack_op = get_linear_prepack_op_for_dtype(weight_dtype)
-                packed_weight = quantizer.quantized_graph.create_node(
-                    'call_function', prepack_op, prepack_args, {})
+                # prepacking weights for static int8 quant and dynamic quant
```

Comment:
for my own understanding, wondering why the code change is for "non-reference option" (line above), but the PR is adding support for reference option?

Reply:
Oh, reference option is supported through modifying `quantize_node` (see previous PR), actually for fp16 static quant both reference and non-reference option would produce the same result

Reply:
here I just added a few dtype checks to make sure the branch conditions are more explicit.